File: java/ReadLineSample/ReadLineSample.java
/* ReadLineSample.java */

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class ReadLineSample {
    public static void main(String... args) {
        InputStreamReader isr = new InputStreamReader(System.in);
        BufferedReader br = new BufferedReader(isr);
        String line;

        System.out.print("Ctrl-D to quit. > ");
        try {
            while ((line = br.readLine()) != null) {
                System.out.println(line);
                System.out.print("Ctrl-D to quit. > ");
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        System.out.println("");
    }
}

File: csharp/ReadLineSample/ReadLineSample.cs
/* ReadLineSample.cs */

using System;

namespace FpigSample
{
  public class ReadLineSample
  {
    public static void Main()
    {
      string line;

      Console.Write("Ctrl-D to quit. > ");
      while ((line = Console.ReadLine()) != null)
      {
        Console.WriteLine(line);
        Console.Write("Ctrl-D to quit. > ");
      }
      Console.WriteLine("");
    }
  }
}

